src/test/py/peer/peer_test.py

```diff
@@ -143,6 +145,41 @@ def test_add_owner(self):
         self.ledger.add_owner(self.peer)
         self.assertEqual(self.ledger.queue[-1][0], self.peer.keys["public"])
 
+    def test_get_owner(self):
+        """
+        Test the get_owner method of Ledger
+        """
+
+        self.ledger.add_owner(self.peer)
+        self.assertEqual(self.ledger.get_owner(), self.peer.keys["public"])
+
+    def test_get_previous_owner(self):
+        """
+        Test the get_previous_owner method of Ledger
+        """
+
+        self.ledger.add_owner(self.peer)
+        self.assertIsNone(self.ledger.get_previous_owner())
+
+        self.ledger.add_owner(self.peer2)
+        self.assertEqual(self.ledger.get_previous_owner(), self.peer.keys["public"])
+
+    def test_get_originator(self):
+        """
+        Test the get_originator method of Ledger
+        """
+
+        self.ledger.add_owner(self.peer)
+        self.assertEqual(self.ledger.get_originator(), self.peer.keys["public"])
+
+    def test_get_owner_history(self):
+        """
+        Test the get_owner_history method of Ledger
+        """
+
+        self.ledger.add_owner(self.peer)
+        self.assertEqual(self.ledger.get_owner_history(), self.ledger.queue)
```

Comment:
To make this test more robust, I would create a local `deque` of the expected contents and `assertEqual()` with that against `get_owner_history()`.